hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/TriggersHandler.java

```diff
@@ -1050,6 +1051,57 @@ public Response getTriggerConditions(
     }
 
     @PUT
+    @Path("/{triggerId}/conditions")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Set the conditions for the trigger. ",
+            notes = "This replaces any existing conditions. Returns the new conditions.",
+            response = Condition.class, responseContainer = "List")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success, Condition Set created."),
+            @ApiResponse(code = 404, message = "No trigger found.", response = ApiError.class),
+            @ApiResponse(code = 500, message = "Internal server error.", response = ApiError.class),
+            @ApiResponse(code = 400, message = "Bad Request/Invalid Parameters", response = ApiError.class)
+    })
+    public Response setConditions(
+            @ApiParam(value = "The relevant Trigger.", required = true)
+            @PathParam("triggerId")
+            final String triggerId,
+            @ApiParam(value = "Collection of Conditions to set.", required = true)
+            final Collection<Condition> conditions) {
+        try {
+            Collection<Condition> updatedConditions = new HashSet<>();
+            conditions.stream().forEach(c -> c.setTriggerId(triggerId));
+            Collection<Condition> firingConditions = conditions.stream()
+                    .filter(c -> c.getTriggerMode() == null || c.getTriggerMode().equals(Mode.FIRING))
+                    .collect(Collectors.toList());
+            if (!isEmpty(firingConditions)) {
+                updatedConditions.addAll(definitions.setConditions(tenantId, triggerId, Mode.FIRING, firingConditions));
+            }
+            Collection<Condition> autoResolveConditions = conditions.stream()
+                    .filter(c -> c.getTriggerMode().equals(Mode.AUTORESOLVE))
+                    .collect(Collectors.toList());
+            if (!isEmpty(autoResolveConditions)) {
```

Comment:
I think we need to still perform the update if the set is empty, otherwise you can't remove all conditions for a particular trigger mode.